plasmapy/formulary/parameters.py

```diff
@@ -284,41 +303,58 @@ def ion_sound_speed(T_e,
 
     ~astropy.units.UnitsWarning
         If units are not provided, SI units are assumed.
+        
+    PhysicsWarning
+        If only one of (k, n_e) is given, the non-dispersive limit 
+        is assumed.
 
     Notes
     -----
-    The ion sound speed :math:`V_S` is approximately given by
+    The ion sound speed :math:`V_S` is given by
 
     .. math::
 
-        V_S = \sqrt{\frac{\gamma_e Z k_B T_e + \gamma_i k_B T_i}{m_i}}
+        V_S = \sqrt{\frac{\gamma_e Z k_B T_e + \gamma_i k_B T_i}{m_i (1 + k^2 \lambda_{D}^2)}}
 
     where :math:`\gamma_e` and :math:`\gamma_i` are the electron and
     ion adiabatic indices, :math:`k_B` is the Boltzmann constant,
     :math:`T_e` and :math:`T_i` are the electron and ion temperatures,
-    :math:`Z` is the charge state of the ion, and :math:`m_i` is the
-    ion mass.
-
-    This function assumes that the product of the wavenumber and the
-    Debye length is small. In this limit, the ion sound speed is not
-    dispersive. In other words, it is frequency independent.
+    :math:`Z` is the charge state of the ion, :math:`m_i` is the
+    ion mass, :math:`\lambda_{D}` is the Debye length, and :math:`k` is the 
+    wavenumber.
+    
+    In the non-dispersive limit (:math:`k^2 \lambda_{D}^2` is small) the 
+    equation for :math:`V_S` is approximated (the denominator reduces 
+    to :math:`m_i`).
 
     When the electron temperature is much greater than the ion
     temperature, the ion sound velocity reduces to
-    :math:`\sqrt{\gamma_e k_B T_e / m_i}`.  Ion acoustic waves can
+    :math:`\sqrt{\gamma_e k_B T_e / m_i}`. Ion acoustic waves can
     therefore occur even when the ion temperature is zero.
 
     Example
     -------
     >>> from astropy import units as u
+    >>> n = 5e19*u.m**-3
+    >>> k_1 = 3e1*u.m**-1
+    >>> k_2 = 3e7*u.m**-1
     >>> ion_sound_speed(T_e=5e6*u.K, T_i=0*u.K, ion='p', gamma_e=1, gamma_i=3)
     <Quantity 203155.0764042 m / s>
-    >>> ion_sound_speed(T_e=5e6*u.K, T_i=0*u.K)
-    <Quantity 203155.0764042 m / s>
-    >>> ion_sound_speed(T_e=500*u.eV, T_i=200*u.eV, ion='D+')
-    <Quantity 229586.01860212 m / s>
+    >>> ion_sound_speed(T_e=5e6*u.K, T_i=0*u.K, n_e=n, k=k_1, ion='p', gamma_e=1, gamma_i=3)
+    <Quantity 203155.03286794 m / s>
+    >>> ion_sound_speed(T_e=5e6*u.K, T_i=0*u.K, n_e=n, k=k_2, ion='p', gamma_e=1, gamma_i=3)
+    <Quantity 310.31329069 m / s>
+    >>> ion_sound_speed(T_e=5e6*u.K, T_i=0*u.K, n_e=n, k=k_1)
+    <Quantity 203155.03286794 m / s>
+    >>> ion_sound_speed(T_e=500*u.eV, T_i=200*u.eV, n_e=n, k=k_1, ion='D+')
+    <Quantity 229585.96150738 m / s>
 
     """
+    
+    if (n_e is None and k is not None) or (k is None and n_e is not None):
+        warnings.warn("The non-dispersive limit has been assumed for "
+                      "this calculation. To prevent this, values must "
+                      "be specified for both n_e and k.", PhysicsWarning)
```

Comment:
Moves this down to be with the `if-else` that determines `klD2`.  This would make the code more readable and better grouped.